server/src/main/java/org/apache/druid/metadata/BasicDataSourceExt.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.metadata;
+
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.apache.commons.dbcp2.ConnectionFactory;
+import org.apache.druid.java.util.common.RE;
+import org.apache.druid.java.util.common.logger.Logger;
+
+import java.sql.Driver;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Properties;
+
+/**
+ * This class exists so that PasswordProvider is asked for password every time a brand new connection is established
+ * with DB. PasswordProvider impls such as based on AWS tokens refresh the underlying token periodically since
+ * each token is valid for a certain period of time only.
+ * So, This class overrides[ummm copies] the methods from base class in order to keep track of connection properties
```

Comment:
Haha, would you fix `ummm copies`?

Reply:
not sure if there is anything to fix as of now , `BasicDataSource` isn't open enough for extension to let us dynamically get password from config everytime a new connection is setup.

Reply:
Oh, I just assumed that you wanted to add a link instead of `ummm copies`. It would be enough to just remove it if there is nothing to fix.

Reply:
i see :) , updated the comment to hopefully remove the link confusion